File: ProblemSolving/BetweenTwoSets/BetweenTwoSets.py
#!/bin/python3

import os
import sys

#
# Complete the getTotalX function below.
#
def getTotalX(a, b):
    #
    # Write your code here.
    #
    count = 0
    for i in range(1, 101):
        atrue = True
        btrue = True
        for num1 in a:
            if i > num1 and i%num1 ==0:
                atrue = True
            elif i%num1 !=0:
                atrue = False
                break
        if atrue:
             for num2 in b:
                 if num2%i!=0:
                        btrue = False
             if btrue:
                 count+=1
    return count
if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    nm = input().split()

    n = int(nm[0])

    m = int(nm[1])

    a = list(map(int, input().rstrip().split()))

    b = list(map(int, input().rstrip().split()))

    total = getTotalX(a, b)

    f.write(str(total) + '\n')

    f.close()

File: ProblemSolving/BetweenTwoSets/BetweenTwoSets.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;

public class Solution {

    /*
     * Complete the getTotalX function below.
     */
    static int getTotalX(int[] a, int[] b) {
        
        int count = 0;
        
        for(int i=1; i<101; i++) {
            
            boolean atrue = true;
            boolean btrue = true;
            
            for(int k=0; k < a.length; k++) {
                
                if(i>= a[k] && i%a[k] == 0) {
                        
                    }
                else
                    atrue = false;
                }
            
            if(atrue) {
                
                for (int l=0; l<b.length; l++) {
                    if (b[l]%i != 0) {
                        btrue = false;
                }
                        
            }
        
                if(btrue)
                    count++;
            }
        }
    return count;
    }

    private static final Scanner scan = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] nm = scan.nextLine().split(" ");

        int n = Integer.parseInt(nm[0].trim());

        int m = Integer.parseInt(nm[1].trim());

        int[] a = new int[n];

        String[] aItems = scan.nextLine().split(" ");

        for (int aItr = 0; aItr < n; aItr++) {
            int aItem = Integer.parseInt(aItems[aItr].trim());
            a[aItr] = aItem;
        }

        int[] b = new int[m];

        String[] bItems = scan.nextLine().split(" ");

        for (int bItr = 0; bItr < m; bItr++) {
            int bItem = Integer.parseInt(bItems[bItr].trim());
            b[bItr] = bItem;
        }

        int total = getTotalX(a, b);

        bw.write(String.valueOf(total));
        bw.newLine();

        bw.close();
    }
}